iggtools/subcommands/build_pangenome.py

```diff
@@ -0,0 +1,180 @@
+            if r_type == 'S':
+                # r itself is the centroid of its cluster
+                centroid_info[r_gene][percent_id] = r_gene
+            elif r_type == 'H':
+                # r is not itself a centroid
+                centroid_info[r_gene][percent_id] = r_centroid
+            else:
+                # ignore all other r types
+                pass
+
+    # At this point we have the max_percent_id centroid for any gene gc, but we lack
+    # coarser clustering assignments for many genes -- we only have those for genes
+    # that are themelves centroids of max_percent_id clusters.
+    #
+    # We can infer the remaining cluster assignments for all genes by transitivity.
+    # For any gene gc, look up the clusters containing gc's innermost centroid,
+    # gc[max_percent_id].  Those clusters also contain gc.
+    percents = cluster_files.keys()
+    max_percent_id = max(percents)
+    for gc in centroid_info.values():
+        gc_recluster = centroid_info[gc[max_percent_id]]
+        for percent_id in percents:
+            gc[percent_id] = gc_recluster[percent_id]
+
+    with OutputStream(gene_info_file) as gene_info:
+        header = ['gene_id'] + [f"centroid_{pid}" for pid in percents]
+        gene_info.write('\t'.join(header) + '\n')
+        genes = centroid_info.keys()
+        for gene_id in sorted(genes):
+            gene_info.write(gene_id)
+            for centroid in centroid_info[gene_id].values():
+                gene_info.write('\t')
+                gene_info.write(centroid)
+            gene_info.write('\n')
+
+
+def build_pangenome(args):
+    """
+    Input spec:  https://github.com/czbiohub/iggtools/wiki#gene-annotations
+    Output spec: https://github.com/czbiohub/iggtools/wiki#pan-genomes
+    """
+
+    species, representatives = read_toc(outputs.genomes)
+
+    species_id = args.species
+    assert species_id in species, f"Species {species_id} is not in the database."
+
+    species_genomes = species[species_id]
+    representative_id = representatives[species_id]
+    species_genomes_ids = species_genomes.keys()
+    tsprint(f"There are {len(species_genomes)} genomes for species {species_id} with representative genome {representative_id}.")
+
+    cleaned = multiprocessing_map(clean_genes, species_genomes_ids)
+
+    command("rm -f genes.ffn")
+    command("rm -f genes.len")
+
+    for temp_files in split(cleaned, 20):  # keep "cat" commands short
+        fna_files, len_files = transpose(temp_files)
+        command("cat " + " ".join(fna_files) + " >> genes.ffn")
+        command("cat " + " ".join(len_files) + " >> genes.len")
+
+    max_percent, lower_percents = CLUSTERING_PERCENTS[0], CLUSTERING_PERCENTS[1:]
+    cluster_files = {max_percent: vsearch(max_percent, "genes.ffn")}
+    recluster = lambda percent_id: vsearch(percent_id, cluster_files[max_percent][0])
+    cluster_files.update(multithreading_hashmap(recluster, lower_percents))
```

Comment:
Here we use threads and not processes, because the CPU intensive work, vsearch, runs in a subcommand.   The python work itself is not intensive, so threads are appropriate.  This would be the common case.  Multiprocessing should be used very rarely.